blocks/bricks/__init__.py

```diff
@@ -345,15 +345,18 @@ def __init__(self, input_dim, output_dim, num_pieces, **kwargs):
         self.output_dim = output_dim
         self.num_pieces = num_pieces
 
-        self.linear_transformation = Linear(
-            name=self.name + '_linear_to_maxout', input_dim=input_dim,
-            output_dim=output_dim * num_pieces, weights_init=self.weights_init,
-            biases_init=self.biases_init, use_bias=self.use_bias)
-        self.maxout_transformation = Maxout(name=self.name + '_maxout',
-                                            num_pieces=num_pieces)
+        self.linear_transformation = Linear(name=self.name +
```

Comment:
What "variety of possible behaviours" are you referring to?

So you're saying that as long as the configuration is limited to passing things down directly, it's okay to push configuration both in `__init__` and in `_push_allocation_config`? That seems a bit arbitrary for me. I would just push configuration down as long as its used by the parent brick (as in my example).

Why wouldn't you want to change the current MLP brick? I find its current behaviour annoying e.g. you can do ```python
>>> mlp.input_dim
50
>>> mlp.push_allocation_config()
>>> mlp.children[0].input_dim = 100
>>> mlp.input_dim == mlp.children[0].input_dim
False
>>> mlp.allocate()
>>> mlp.apply(x).eval({x: np.random.rand(10, mlp.input_dim)})  # Will crash 
>>> mlp.apply(x).eval({x: np.random.rand(10, mlp.children[0].input_dim)})  # Won't
```